### PR TITLE
[SC] Matured blocks improvements

### PR DESCRIPTION
**Pull request by fassadlr**
General logging and result returning improvements around the mature block deposit provider. The main changes being that the controller now returns a full result class. 

I am also still testing this with the test tool.

This PR can be reviewed commit by commit.

### CODE REVIEW

**Comment by fassadlr** [comment]
I have now tested these changes via the test tool and is well.

File: src/Stratis.Features.FederatedPeg/Controllers/FederationGatewayController.cs [hunk 1]
```diff
@@ -71,16 +71,8 @@ public async Task<IActionResult> GetMaturedBlockDepositsAsync([FromBody] Matured
...
-                    blockRequest.BlockHeight, blockRequest.MaxBlocksToSend).ConfigureAwait(false);
-
-                if (depositsResult.IsSuccess)
-                {
-                    return this.Json(depositsResult.Value);
-                }
-
-                this.logger.LogTrace("Error calling /api/FederationGateway/{0}: {1}.", FederationGatewayRouteEndPoint.GetMaturedBlockDeposits, depositsResult.Error);
-                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, $"Could not re-sync matured block deposits: {depositsResult.Error}", depositsResult.Error);
+                SerializableResult<List<MaturedBlockDepositsModel>> depositsResult = this.maturedBlocksProvider.GetMaturedDeposits(blockRequest.BlockHeight, blockRequest.MaxBlocksToSend);
```
Comment (codingupastorm) [inline src/Stratis.Features.FederatedPeg/Controllers/FederationGatewayController.cs]: Nice.

File: src/Stratis.Bitcoin/SerializableResult.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+namespace Stratis.Bitcoin
+{
+    /// <summary>
+    /// A generic result type that can be serialized.
+    /// </summary>
+    /// <typeparam name="T">The type of the value to return if the result was successful.</typeparam>
+    public sealed class SerializableResult<T>
+    {
+        [JsonProperty("isSuccess")]
+        public bool IsSuccess { get; private set; }
```
Comment (codingupastorm) [inline src/Stratis.Bitcoin/SerializableResult.cs]: Useful class, this property isn't used at the moment but that's fine

**Comment by fassadlr** [comment]
https://stratisplatformuk.visualstudio.com/StratisBitcoinFullNode/_workitems/edit/4069

**Comment by codingupastorm** [comment]
Want to fix the conflicts @fassadlr ? And this can go in

**Comment by fassadlr** [comment]
I have tested these changes extensively on the test tool with deposits and withdrawals.

**Pull request merged**